visualization/include/pcl/visualization/point_picking_event.h

```diff
@@ -71,21 +77,24 @@ namespace pcl
         performSinglePick (vtkRenderWindowInteractor *iren, float &x, float &y, float &z);
 
         int
-        performAreaPick (vtkRenderWindowInteractor *iren, std::vector<int> &indices) const;
+        performAreaPick (vtkRenderWindowInteractor *iren,
+                         CloudActorMapPtr cam_ptr,
+                         std::map<std::string, Indices>& cloud_indices) const;
+
 
       private:
         float x_, y_, z_;
         int idx_;
         bool pick_first_;
+        const vtkActor* actor_;
```

Comment:
This shoud be changed to const vtkSmartPointer<vtkActor> actor_;
to get right value of pointer based on @vigneshsara request

Reply:
I don't see how changing it to a smart pointer fixes that issue? 

The use of a smart pointer vs a raw pointer is a matter of ownership. With the smart pointer you're taking shared ownership of the pointer and you're saying that you are responsible for freeing the object. With the raw pointer you're not interested in deallocating or keeping the object alive. You're just storing a reference. 

An event should not take ownership of a vtkActor, as the event will only exist as long as the actor exists. It's fired once and then it's destroyed.

The problem @vigneshsara is mentioning, which reads like an incorrect pointer comparison is taking place, is agnostic to the use of smart or raw pointers.

Reply:
actor_   values return  from vtk ,that not available in cam_ptr map second values so that return the empty value in cloud  name

Reply:
actor_   values return  from vtk ,that not available in cam_ptr map second values so that return the empty value in cloud  name

Reply:
actor_   values return  from vtk ,
that not available in cam_ptr map second values so that return the empty value in cloud  name